### PR TITLE
fixing bug in get_scf_accuracy tool in pw

### PR DESCRIPTION
**Pull request by d-tomerini**
latest commit introduced a tool to retrieve the ``scf_accuracy``, but
there was a problem with the -1 index, fixed here

### CODE REVIEW

**Comment by d-tomerini** [comment]
Apologies for my sloppy review the other day, @sphuber. I should have tested it in a real calculation

**Comment by d-tomerini** [comment]
Now I'm wondering why the test does not fail, as it is on the array content. I'm checking and get back to you

**Comment by d-tomerini** [comment]
I agree. I was thinking to use the sanity check we perform in the parser to actually build the ``scf_accuracy_index``, so it should just be a few lines more in the function and use it.

I'll add it to this PR

**Comment by d-tomerini** [comment]
@sphuber ,
I made the necessary changes, adapted the pw raw parser, the get_scf_accuracy function and the tests

**Comment by d-tomerini** [comment]
Sorry, I've made a mess here

File: aiida_quantumespresso/tools/calculations/pw.py [hunk 1]
```diff
@@ -34,16 +34,25 @@ def get_scf_accuracy(self, index=0):
...
+            raise IndexError(
+                'invalid index {}, must be between {} and {}'.format(index, -number_of_frames, number_of_frames - 1)
+            )
+
+        # building an scf_accuracy_index for easier manipulation
+        scf_accuracy_index = [0]
+        value = 0
+        for i in scf_iterations:
+            value += i
+            scf_accuracy_index.append(value)
```
Comment (sphuber) [inline aiida_quantumespresso/tools/calculations/pw.py]: How about this instead?
```
for i in scf_iterations:
    scf_accuracy_index.append(scf_accuracy_index[-1] + i)
```

File: aiida_quantumespresso/parsers/parse_raw/pw.py [hunk 2]
```diff
@@ -835,19 +831,14 @@ def parse_stdout(stdout, input_parameters, parser_options=None, parsed_xml=None)
...
-            logs.warning.append(warning)
-        value = trajectory_data['scf_accuracy_index'][0]
-        for i, j in zip(trajectory_data['scf_iterations'], trajectory_data['scf_accuracy_index'][1:]):
-            value += i
-            if value != j:
+        if 'scf_iterations' in trajectory_data:
+            if len(trajectory_data['scf_accuracy']) != sum(trajectory_data['scf_iterations']):
                 logs.warning.append(warning)
+        else:
+            logs.warning.append(warning)
```
[the remark this answers is not in the document]
Comment (d-tomerini) [reply aiida_quantumespresso/parsers/parse_raw/pw.py]: good suggestion

**Pull request merged**